packages/integration-tests/test/000_withdrawals.spec.ts

```diff
@@ -0,0 +1,212 @@
+// Named 000 in order to run first since the output submitter
+// can fall behind.
+
+/* Imports: External */
+import { BigNumber, constants, Contract, ContractReceipt, utils, Wallet } from 'ethers'
+import { awaitCondition } from '@eth-optimism/core-utils'
+import * as rlp from 'rlp'
+import { Block } from '@ethersproject/abstract-provider'
+
+/* Imports: Internal */
+import { WITHDRAWER_ADDR } from './shared/constants'
+import env from './shared/env'
+import { expect } from './shared/setup'
+import winston from 'winston'
+
+const withdrawerArtifact = require('../../contracts/artifacts/contracts/L2/Withdrawer.sol/Withdrawer.json')
+const l2OOracleArtifact = require('../../contracts/artifacts/contracts/L1/L2OutputOracle.sol/L2OutputOracle.json')
+const counterArtifact = require('../artifacts/Counter.sol/Counter.json')
+const multiDepositorArtifact = require('../artifacts/MultiDepositor.sol/MultiDepositor.json')
+
+describe('Withdrawals', () => {
+  let logger: winston.Logger
+  let portal: Contract
+  let withdrawer: Contract
+
+  let recipient: Wallet
+
+  before(async () => {
+    logger = env.logger
+    portal = env.optimismPortal
+
+    withdrawer = new Contract(
+      WITHDRAWER_ADDR,
+      withdrawerArtifact.abi,
+    )
+  })
+
+  describe('simple withdrawals', () => {
+    let nonce: BigNumber
+    let burnBlock: Block
+    let withdrawalHash: string
+    const value = utils.parseEther('1')
+    const gasLimit = 3000000
+
+    before(async function () {
+      this.timeout(60_000)
+      recipient = Wallet.createRandom().connect(env.l2Provider)
+      withdrawer = withdrawer.connect(recipient)
+
+      logger.info('Generated new wallet', {
+        recipient: recipient.address,
+      })
+      logger.info('Depositing to new address on L2')
+      let tx = await portal.connect(env.l1Wallet).depositTransaction(
+        recipient.address,
+        utils.parseEther('1.337'),
+        gasLimit,
+        false,
+        [],
+        {
+          value: utils.parseEther('1.337'),
+        },
+      )
+      await tx.wait()
+
+      await awaitCondition(async () => {
+        const bal = await recipient.getBalance()
+        return bal.eq(tx.value)
```

Comment:
This isn't stateless, ie you would need to bring down the system and bring it up again for this condition to be met. It would be better to do something like what the hardhat task does, where it does `DepositTx.fromL1Event().hash()` and then polls L2 for that tx hash

Reply:
What do you mean? This condition does get met, otherwise the tests wouldn't pass. The recipient is a random address with zero balance, so we can assert that the balance equals the transaction's value.

I can update it to use the same thing the hardhat task does, assuming those libraries are accessible to the integration tests package. Otherwise, I'd like to do that cleanup once we've moved into the monorepo. I'd rather not invest time into making code reuse work if we're just going to rip all that out in a week or two.

Reply:
Ah got it - the random address part that the value is sent to prevents the problem that I was thinking of, my bad. I was thinking running the tests twice against the same state but since its a random address its fine

Reply:
Ok sounds good!